examples/usages/context_menu.rs

```diff
@@ -0,0 +1,190 @@
+};
+use std::fmt::Debug;
+
+#[derive(Event)]
+struct OpenContextMenu {
+    pos: Vec2,
+}
+
+#[derive(Event)]
+struct CloseContextMenus;
+
+#[derive(Component)]
+struct ContextMenu;
+
+#[derive(Component)]
+struct ContextMenuItem(Srgba);
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_observer(on_trigger_menu)
+        .add_observer(on_trigger_close_menus)
+        .add_observer(create_hover_observer::<Out>(basic::WHITE))
+        .add_observer(create_hover_observer::<Over>(basic::RED))
+        .run();
+}
+
+fn create_hover_observer<T: Debug + Clone + Reflect>(
+    color: Srgba,
+) -> impl FnMut(Trigger<Pointer<T>>, Query<&mut TextColor>, Query<&Children>) {
+    move |trigger: Trigger<Pointer<T>>,
+          mut query: Query<&mut TextColor>,
+          children: Query<&Children>| {
+        let Ok(children) = children.get(trigger.target()) else {
+            return;
+        };
+
+        for child in children.iter() {
+            if let Ok(mut col) = query.get_mut(child) {
+                col.0 = color.into();
+            }
+        }
+    }
+}
+
+fn setup(mut commands: Commands) {
+    commands.spawn(Camera2d);
+
+    commands.spawn(button()).observe(
+        // any click bubbling up here should lead to closing any open menu
+        |_trigger: Trigger<Pointer<Pressed>>, mut commands: Commands| {
+            commands.trigger(CloseContextMenus);
+        },
+    );
+}
+
+fn on_trigger_close_menus(
+    _trigger: Trigger<CloseContextMenus>,
+    mut commands: Commands,
+    menus: Query<Entity, With<ContextMenu>>,
+) {
+    for e in menus.iter() {
+        commands.entity(e).despawn();
+    }
+}
+
+fn on_trigger_menu(trigger: Trigger<OpenContextMenu>, mut commands: Commands) {
+    commands.trigger(CloseContextMenus);
+
+    let pos = trigger.pos;
+
+    debug!("open context menu at: {pos}");
+
+    commands
+        .spawn((
+            Name::new("context menu"),
+            ContextMenu,
+            Node {
+                position_type: PositionType::Absolute,
+                left: Val::Px(pos.x),
+                top: Val::Px(pos.y),
+                flex_direction: FlexDirection::Column,
+                ..default()
+            },
+            BorderColor(Color::BLACK),
+            BorderRadius::all(Val::Px(4.)),
+            BackgroundColor(Color::linear_rgb(0.1, 0.1, 0.1)),
+            children![
+                context_item("fuchsia", basic::FUCHSIA),
+                context_item("gray", basic::GRAY),
+                context_item("maroon", basic::MAROON),
+                context_item("purple", basic::PURPLE),
+                context_item("teal", basic::TEAL),
+            ],
+        ))
+        .observe(
+            |trigger: Trigger<Pointer<Pressed>>,
+             query: Query<&ContextMenuItem>,
+             mut clear_col: ResMut<ClearColor>,
+             mut commands: Commands| {
+                let target = trigger.event().target;
```

Comment:
Nit: Should we add a comment about how here we're not relying on `trigger.target` but `trigger.event().target`, and take that opportunity to explain the difference ?

Reply:
Related discussion about this API on discord: https://discord.com/channels/691052431525675048/692572690833473578/1328737758583455875

Reply:
added a comment here to clarify